test/test_api/test_api.py

```diff
@@ -449,3 +551,59 @@ def test_do_dummy_prediction(dask_client, fit_dictionary_tabular):
     estimator._clean_logger()
 
     del estimator
+
+
+@pytest.mark.parametrize('include_traditional', (True, False))
+def test_get_incumbent_results(dataset_traditional_classifier_num_only, backend, include_traditional):
+    X, y = dataset_traditional_classifier_num_only
+
+    X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(
+        X, y, random_state=1)
+
+    # Search for a good configuration
+    estimator = TabularClassificationTask(
+        backend=backend,
+        resampling_strategy=HoldoutValTypes.holdout_validation,
+    )
+
+    InputValidator = TabularInputValidator(
+        is_classification=True,
+    )
+
+    # Fit a input validator to check the provided data
+    # Also, an encoder is fit to both train and test data,
+    # to prevent unseen categories during inference
+    InputValidator.fit(X_train=X_train, y_train=y_train, X_test=X_test, y_test=y_test)
+
+    dataset = TabularDataset(
+        X=X_train, Y=y_train,
+        X_test=X_test, Y_test=y_test,
+        validator=InputValidator,
+        resampling_strategy=estimator.resampling_strategy,
+        resampling_strategy_args=estimator.resampling_strategy_args,
+    )
+
+    pipeline_run_history = RunHistory()
+    pipeline_run_history.load_json('test/test_api/.tmp_api/runhistory.json', estimator.get_search_space(dataset))
+
+    estimator._do_dummy_prediction = unittest.mock.MagicMock()
+
+    with unittest.mock.patch.object(AutoMLSMBO, 'run_smbo') as AutoMLSMBOMock:
+        with unittest.mock.patch.object(TabularClassificationTask, '_do_traditional_prediction',
+                                        new=dummy_traditional_classification):
+            AutoMLSMBOMock.return_value = (pipeline_run_history, {}, 'epochs')
+            estimator.search(
+                X_train=X_train, y_train=y_train,
+                X_test=X_test, y_test=y_test,
+                optimize_metric='accuracy',
+                total_walltime_limit=150,
+                func_eval_time_limit_secs=50,
+                enable_traditional_pipeline=True,
+                load_models=False,
+            )
+    config, results = estimator.get_incumbent_results(include_traditional=include_traditional)
```

Comment:
is it possible to check that the returned config has the lowest cost?

Reply:
sure